backend/loadbalancer.go

```diff
@@ -22,7 +24,10 @@ type LoadBalancer struct {
 
 // NewLoadBalancer creates a new instance of LoadBalancer with the given backends.
 // It takes a slice of RequestHandler as a parameter and returns a new instance of LoadBalancer.
-func NewLoadBalancer(backends []wasabi.RequestHandler) (*LoadBalancer, error) {
+func NewLoadBalancer(backends []struct {
+	handler wasabi.RequestHandler
+	weight  uint
+}) (*LoadBalancer, error) {
```

Comment:
this is not usable outside of backend package, please define public type for this with public fields.

Reply:
fixed